src/VibrosonicsAPI.cpp

```diff
@@ -21,34 +21,44 @@ void VibrosonicsAPI::init()
  */
 void VibrosonicsAPI::processAudioInput(float output[])
 {
+    // Copy samples from rolling buffer to FFT input array
+    for (int i = 0; i < WINDOW_SIZE_OVERLAP; i++) {
+      vData[i] = rollingInputBuffer[i];
+    }
+
     // Use Fast4ier combined with Vibrosonics FFT functions
     dcRemoval();
     fftWindowing();
-    Fast4::FFT(vData, WINDOW_SIZE);
+    Fast4::FFT(vData, WINDOW_SIZE_OVERLAP);
     complexToMagnitude();
 
     // Copy complex data to float arrays
-    for (int i = 0; i < WINDOW_SIZE; i++) {
+    for (int i = 0; i < WINDOW_SIZE_BY_2; i++) {
         vReal[i]  = vData[i].re();
         output[i] = vReal[i];
     }
+
+    // Move rolling buffer forward to make room for next set of samples
+    for (int i = WINDOW_SIZE; i < WINDOW_SIZE_OVERLAP; i++) {
+      rollingInputBuffer[i - WINDOW_SIZE] = rollingInputBuffer[i];
+    }
```

Comment:
Minor Suggestion: I think we can use the same idea of copying the entire buffer at once rather than using a for loop.


```suggestion
    for (int i = WINDOW_SIZE; i < WINDOW_SIZE_OVERLAP; i++) {
      rollingInputBuffer[i - WINDOW_SIZE] = rollingInputBuffer[i];
    }
    std::copy(rollingInputBuffer + WINDOW_SIZE, rollingInputBuffer + WINDOW_SIZE_OVERLAP, rollingInputBuffer);
```